src/components/Microsoft.DotNet.UpgradeAssistant.Extensions/ExtensionManager.cs

```diff
@@ -55,6 +65,17 @@ public sealed class ExtensionManager : IDisposable, IExtensionManager
                     LoadPath(path, isDefault: true);
                 }
 
+                try
+                {
+                    foreach (var path in Registered.Select(_extensionDownloader.GetInstallPath))
```

Comment:
Probably a good idea to move the `Registered.Select(_extensionDownloader.GetInstallPath)` out of the forloop and be a variable so it doesnot need to resolve it in every loop

Reply:
Which loop are you refering to? The initializer is only invoked once (see https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/statements/iteration-statements#the-for-statement) if that's what you're referring to

Reply:
`var s = Registered.Select(_extensionDownloader.GetInstallPath);
foreach (var path in s)
{
...
}`

Reply:
Not needed - initializers are only evaluated once (see the linked spec)